modules/reactor/include/meshgenerators/PinMeshGenerator.h

```diff
@@ -32,49 +34,68 @@ class PinMeshGenerator : public ReactorGeometryMeshBuilderBase
   // Define metadata associated with PinMeshGenerator
   void generateMetadata();
 
-  ///The id number for this pin type
+  /**
+   * Get CSGSurfaces corresponding to hexagonal or square region with given halfpitch
+   * @param radial_index Radial index of hex / square region, for surface naming
+   * @param halfpitch Halfpitch of square or hexagon
+   * @param csg_obj Reference to CSGBase object for adding defined surfaces to
+   * @return vector of surfaces that correspond to hexagonal or square region
+   */
+  std::vector<std::reference_wrapper<const CSG::CSGSurface>> getHexCartSurfaces(
+      unsigned int radial_index, Real halfpitch, std::unique_ptr<CSG::CSGBase> & csg_obj);
+
+  /**
+   * Get inner region of given surfaces, defined as the intersection
+   * of halfspaces of each surface. Here, the halfspace direction is determined based on the origin.
+   * @param radial_surfaces List of references to surfaces used to define inner region
+   * @return inner region defined by surfaces
+   */
+  CSG::CSGRegion getInnerRegion(
```

Comment:
this seems more general than just the pin ?

Reply:
just needs an extra parameter for the origin in the general case

Reply:
> Are you suggesting the additional parameter to specify the origin should control how the surfaces are translated to move the center to another location?

no the origin is just used to determine which half space we are in for each of the surface

no translations or rotations involved, just a point and bunch of surfaces. it s a general need

> Also, what is the purpose of specifying that the minimum number of surfaces should be used to define radial_surfaces?

if you only use the minimum number of surfaces, it s more efficient in the downstream use imo. 
It s like advice

Reply:
Got it, the origin is used to determine direction of halfspace given the surfaces are already translated correct?